packages/manager/modules/iam/src/components/actionSelect/actionSelect.controller.js

```diff
@@ -0,0 +1,303 @@
+import { cloneDeep } from 'lodash-es';
+import ActionTrees from './ActionTrees.class';
+
+export default class ActionSelectController {
+  /* @ngInject */
+  constructor($attrs, $scope, $timeout, $translate, ReferenceService) {
+    this.$attrs = $attrs;
+    this.$scope = $scope;
+    this.$timeout = $timeout;
+    this.$translate = $translate;
+    this.ReferenceService = ReferenceService;
+
+    /**
+     * List of actions
+     * @type {Object[]}
+     */
+    this.actions = [];
+
+    /**
+     * Group each action into categories and each category into resource types
+     * So it can be displayed as collapsible components
+     * @type {ActionTrees=}
+     */
+    this.actionTrees = undefined;
+
+    /**
+     * The custom action's model
+     * @type {string}
+     */
+    this.customActionModel = '';
+
+    /**
+     * The custom action's success message, if any
+     * @type {string}
+     */
+    this.customActionSuccessMessage = '';
+
+    /**
+     * The NgFormController created by the template by using name="$ctrl.form"
+     * @type {NgFormController}
+     */
+    this.form = null;
+
+    /**
+     * Whether the controller is loading
+     * @type {boolean}
+     */
+    this.isLoading = false;
+
+    /**
+     * Whether all the wildcard switch is active
+     * @type {boolean}
+     */
+    this.isWildcardActive = false;
+
+    /**
+     * The main form's name
+     * @type {string}
+     */
+    this.name = 'actions';
+
+    /**
+     * Whether the select is required
+     * @type {boolean}
+     */
+    this.required = false;
+
+    /**
+     * The input size attribute
+     * @type {string}
+     */
+    this.size = 'xl';
+
+    /**
+     * The snapshot of the user selection
+     * @type {ActionTrees=}
+     */
+    this.snapshot = undefined;
```

Comment:
`undefined` is used so the property can be destructured with a default value (`null` is not considered as optional)